pandas/core/frame.py

```diff
@@ -7994,6 +8018,47 @@ def blk_func(values):
             result = self._constructor_sliced(result, index=labels)
         return result
 
+    def _reduce_columns(self, op, name, skipna=True, **kwds):
+        """
+        Reduce DataFrame column-wise.
+
+        Parameters
+        ----------
+        op : func
+            The reducing function to be called on the values. Only used
+            for columns backed by a numpy ndarray.
+        name : str
+            The name of the reduction.
+        skipna, **kwds : keywords to pass to the `op` function
+
+        Returns
+        -------
+        Series
+        """
+        result = []
+
+        for arr in self._iter_arrays():
+            if isinstance(arr, ExtensionArray):
```

Comment:
could this go inside `blk_func`?  i think that would avoid the need for a bunch of edits.

Reply:
Is suppose so, yes, I just commented the same above

Reply:
done